tests/core/utilities/test_construct_event_topics.py

```diff
@@ -31,48 +31,46 @@ def hex_and_pad(i):
         (
             EVENT_1_ABI,
             {},
-            [[EVENT_1_TOPIC]],
```

Comment:
This API change doesn't have any downstream effects in the public API?

Reply:
There shouldn't be a bad effect. Because the topic list permutations were removed, `construct_event_topics` only needs to return a nested list when there are multiple options for a single argument filter.

There was a bug in how the topic list was being generated.  This is an example of a correctly formated topic list to match logs with either of two values for the first indexed event arg:

```
[ event_sig_topic, [option1_for_arg1, option2_for_arg1], arg2_topic ]
```

whereas this is what we were generating to be equivalent to the above; we would pass full topic list permutation against each option:
```
[
    [ event_sig_topic, option1_for_arg1, arg2_topic ],
    [ event_sig_topic, option1_for_arg1, arg2_topic ]
]
```
This topic topic list actual means: any of `[event_sig_topic, option1_for_arg1, arg2_topic ]` in the first position and any of `[event_sig_topic, option1_for_arg1, arg2_topic]` in the second position.  Node filters don't understand nested lists as full topic lists.  The spec defines a single topic list that can have nested lists of options for each positional element.

Reply:
I can break this part out into another PR, and add a failing test.

This issue has been fixed in the new api: https://github.com/ethereum/web3.py/pull/953.  Im fixing this function to make it easier to integrate the new filter builder api.